compiler/rustc_const_eval/src/interpret/operand.rs

```diff
@@ -213,7 +213,23 @@ impl<Prov: Provenance> std::fmt::Display for ImmTy<'_, Prov> {
                     write!(f, "{:x}: {}", s, self.layout.ty)
                 }
                 Immediate::ScalarPair(a, b) => {
-                    // FIXME(oli-obk): at least print tuples and slices nicely
+                    // Try to print slices nicely first
+                    // It's *only* printed for internal debugging
+                    if let Some(ty) = tcx.lift(self.layout.ty)
+                        && let ty::Slice(element_ty) = ty.kind()
+                    {
+                        // For slices, the first scalar is the pointer, second is the length
+                        let ptr_str = FmtPrinter::print_string(tcx, Namespace::ValueNS, |cx| {
+                            p(cx, a, Ty::new_ptr(tcx, *element_ty, ty::Mutability::Not))
+                        })?;
+                        let len_str = FmtPrinter::print_string(tcx, Namespace::ValueNS, |cx| {
+                            p(cx, b, tcx.types.usize)
+                        })?;
+
+                        write!(f, "&[{}; {}]: {}", ptr_str, len_str, ty)?;
```

Comment:
Do you have an example of what this looks like in the end?

It seems like it'll print something like `&[ptr; 128]` which is confusing because it looks like a slice repeating the pointer 128 times, when actually it is a slice of that length starting at `ptr`.